packages/bee-q/src/components/notification/bq-notification.tsx

```diff
@@ -0,0 +1,254 @@
+  // ===================================
+
+  @Element() el!: HTMLBqNotificationElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  /** If true, the notification will automatically hide after the specified amount of time */
+  @Prop({ reflect: true }) autoDismiss: boolean;
+
+  /** If true, the close button at the top right of the notification won't be shown */
+  @Prop({ reflect: true }) disableClose: boolean;
+
+  /** If true, the predefined icon type won't be shown and a custom icon provided on integration will be displayed instead */
+  @Prop({ reflect: true }) hasCustomIcon: boolean;
+
+  /** If true, the notification icon won't be shown */
+  @Prop({ reflect: true }) hideIcon: boolean;
+
+  /** If true, the notification will be shown */
+  @Prop({ reflect: true, mutable: true }) isOpen: boolean;
+
+  /** The length of time, in milliseconds, after which the notification will close itself. Only valid if `autoDismiss="true"` */
+  @Prop({ reflect: true }) time: number = 3000;
+
+  /** Type of Notification */
+  @Prop({ reflect: true }) type: TNotificationType = 'info';
+
+  // Prop lifecycle events
+  // =======================
+  @Watch('autoDismiss')
+  @Watch('time')
+  handleTimeout() {
+    this.autoDismissDebounce?.cancel();
+    if (!this.autoDismiss) return;
+
+    this.autoDismissDebounce = debounce(() => {
+      this.hide();
+    }, this.time);
+    // Make sure to autodismiss the notification if the `auto-dismiss` value changed while open
+    if (this.isOpen) this.autoDismissDebounce();
+  }
+
+  @Watch('isOpen')
+  handleOpenChange() {
+    this.autoDismissDebounce?.cancel();
+
+    if (!(this.autoDismiss && this.isOpen)) return;
+    this.autoDismissDebounce();
+  }
+
+  @Watch('type')
+  checkPropValues() {
+    validatePropValue(NOTIFICATION_TYPE, 'info', this.el, 'type');
+  }
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /** Callback handler to be called when the notification is hidden */
+  @Event() bqHide: EventEmitter;
+
+  /** Callback handler to be called when the notification is shown */
+  @Event() bqShow: EventEmitter;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  componentWillLoad() {
+    this.checkPropValues();
+    this.handleTimeout();
+  }
+
+  // Listeners
+  // ==============
+
+  @Listen('bqHide')
+  onNotificationHide() {
+    try {
+      notificationPortal.removeChild(this.el);
+      // Remove the notification portal from the DOM when there are no more notifications
+      if (notificationPortal.querySelector('bq-notification') === null) {
+        notificationPortal.remove();
+      }
+    } catch (error) {
+      if (error instanceof DOMException) return;
+      console.error(error);
```

Comment:
We can also `throw error` here and let the default behaviour. Not sure if is better though